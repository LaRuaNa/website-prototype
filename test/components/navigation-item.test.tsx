import React from 'react';
import renderer from 'react-test-renderer';
import NavigationItem from '../../src/components/navigation-item';
import { createNavigationItemData } from '../__fixtures__/navigation';

describe('NavigationItem component', () => {
  it('renders correctly', () => {
    const {
      key,
      title,
      slug,
      isRead,
      isActive,
      onClick,
      autoScroll,
    } = createNavigationItemData();
    const tree = renderer
      .create(
        <NavigationItem
          key={key}
          title={title}
          slug={slug}
          isRead={isRead}
          isActive={isActive}
          onClick={onClick}
          autoScroll={autoScroll}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
