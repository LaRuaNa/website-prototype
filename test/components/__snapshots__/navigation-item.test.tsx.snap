// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem component renders correctly 1`] = `
<a
  className="t-body2 side-nav__item side-nav__item--done"
  href="/learn/"
  innerRef={[Function]}
  onClick={[Function]}
>
  
</a>
`;
